drivers.py

```diff
@@ -92,11 +102,151 @@ def upload_file(self, src, obj_path):
         return dest
 
 
+class GoogleDriveDriver(Driver):
+    """Driver to handle connection to Google Drive"""
+
+    def __init__(self, config):
+        super(GoogleDriveDriver, self).__init__(config)
+
+        try:
+            self._credentials = service_account.Credentials.from_service_account_file(
+                Path(config.google_drive.service_account_file),
+                scopes=["https://www.googleapis.com/auth/drive.file"],
+            )
+
+            self._service: Resource = build(
+                "drive", "v3", credentials=self._credentials
+            )
+
+            self._folder = config.google_drive.folder
+            folder_exists, self._folder_id = self._folder_exists(self._folder)
+
+            if not folder_exists:
+                self._folder_id = self._create_folder(self._folder)
+
+            for email in get_share_with(config.google_drive):
+                if email:
+                    self._share_with(email)
+
+        except Exception as e:
+            raise DriverError("GoogleDrive driver init error: " + str(e))
+
+    def upload_file(self, src: str, obj_path: str):
+        try:
+            file_metadata = {
+                "name": obj_path,
+                "parents": [self._folder_id],
+            }
+            media = MediaFileUpload(src)
+
+            file = (
+                self._service.files()
+                .create(body=file_metadata, media_body=media, fields="id")
+                .execute()
+            )
+
+            file_id = file.get("id")
+
+        except Exception as e:
```

Comment:
Is there a reason for such broad exception? I can see in many places call of some `.execute()` method which might have a well defined exception?

Reply:
Most of those **googleapiclient** functions can fail in quite several ways, it seemed better to catch all types of error that can happen there. Otherwise it could happen that we will miss some type of uncommon error.

Reply:
ok, fair enough